bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -111,8 +111,12 @@ public void login() throws MeaterException {
             json = httpResponse.getContentAsString();
             JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
             JsonObject childObject = jsonObject.getAsJsonObject("data");
-            this.authToken = childObject.get("token").getAsString();
-            this.userId = childObject.get("userId").getAsString();
+            JsonElement tokenJson = childObject.get("token");
+            if (tokenJson != null) {
+                this.authToken = tokenJson.getAsString();
+            } else {
+                throw new MeaterException("Tken JSON is null");
```

Comment:
Typo: Token instead of Tken.
Maybe a better message would be "Token is not present in the JSON response" ?

Reply:
Agree and fixed!